procfile/extractor.go

```diff
@@ -0,0 +1,191 @@
+package procfile
```

Comment:
I pulled out the stuff we were doing in resolver.go and extractor.go into a new package for dealing with procfiles.